src/reanimated2/animation/clamp.ts

```diff
@@ -0,0 +1,102 @@
+'use strict';
+import { defineAnimation, getReduceMotionForAnimation } from './util';
+import type {
+  Animation,
+  Timestamp,
+  AnimatableValue,
+  AnimationObject,
+  ReduceMotion,
+} from '../commonTypes';
+import type { ClampAnimation } from './commonTypes';
+
+type withClampType = <T extends AnimatableValue>(
+  config: { min?: number; max?: number; reduceMotion?: ReduceMotion },
+  clampedAnimation: T
+) => T;
+
+// TODO This feature is not documented yet
+export const withClamp = function <T extends AnimationObject<number>>(
+  config: { min?: number; max?: number; reduceMotion?: ReduceMotion },
+  _clampedAnimation: T | (() => T)
+): Animation<ClampAnimation> {
+  'worklet';
+  return defineAnimation<ClampAnimation, T>(
+    _clampedAnimation,
+    (): ClampAnimation => {
+      'worklet';
+      const clampedAnimation =
+        typeof _clampedAnimation === 'function'
+          ? _clampedAnimation()
+          : _clampedAnimation;
+
+      function clampOnFrame(
+        animation: ClampAnimation,
+        now: Timestamp
+      ): boolean {
+        const finished = clampedAnimation.onFrame(clampedAnimation, now);
+
+        if (clampedAnimation.current === undefined) {
+          // This should never happen
+          // TODO check if nextAnimation.current should be optional
```

Comment:
Instead of just adding `// TODO`, let's do it 😅

Reply:
You're right I checked, and this should required. However I don't want to change anything within this PR cause I know that @tjzel is still working on typing our animations, so I'll just add a comment with information

Reply:
This is hard 😵